src/Database/LSMTree/Internal/WriteBufferBlobs.hs

```diff
@@ -226,25 +226,27 @@ readBlob fs WriteBufferBlobs {blobFileState} BlobSpan {blobSpanOffset, blobSpanS
         error "BlobFile.mkBlobRef: no such blob refs, no blob file open"
 
 -- | Helper function to make a 'BlobRef' that points into a 'WriteBufferBlobs'.
+-- TODO: this two stage function design isn't terribly nice, and allocates
+-- more than necessary (a closure every call).
 mkBlobRef :: PrimMonad m
           => WriteBufferBlobs m h
-          -> BlobSpan
-          -> m (BlobRef m (FS.Handle h))
-mkBlobRef WriteBufferBlobs {blobFileState, blobFileRefCounter} blobRefSpan = do
+          -> m (BlobSpan -> BlobRef m (FS.Handle h))
+mkBlobRef WriteBufferBlobs {blobFileState, blobFileRefCounter} = do
```

Comment:
In one place where it's used, it's much more convenient for the conversion to be a pure function.

But I think the whole set of conversions could be done better. So I thought it might be better to resolve it as part of doing that.